From df19c5a597991942bb771df7f6450b6eb15f882a Mon Sep 17 00:00:00 2001
From: Elliott Sales de Andrade <quantum.analyst@gmail.com>
Date: Sat, 16 Sep 2017 05:02:57 -0400
Subject: [PATCH 8/9] Use strtoll when reading mtime from build log.

This prevents overflow on Windows where 'long' is not 64-bit.
---
 src/build_log.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/build_log.cc b/src/build_log.cc
index a591050..648617c 100644
--- a/src/build_log.cc
+++ b/src/build_log.cc
@@ -290,7 +290,7 @@ bool BuildLog::Load(const string& path, string* err) {
     if (!end)
       continue;
     *end = 0;
-    restat_mtime = atol(start);
+    restat_mtime = strtoll(start, NULL, 10);
     start = end + 1;
 
     end = (char*)memchr(start, kFieldSeparator, line_end - start);
-- 
2.14.1

